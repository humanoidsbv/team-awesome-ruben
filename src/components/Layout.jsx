/* eslint-disable react/jsx-filename-extension */
import React from 'react';
import PropTypes from 'prop-types';

import styles from './Layout.module.css';

const Layout = ({ children }) => (
  <main className={styles.layout}>{children}</main>
);

Layout.propTypes = {
  children: PropTypes.oneOfType([
    PropTypes.arrayOf(PropTypes.node),
    PropTypes.node
  ]).isRequired
};

export default Layout;
