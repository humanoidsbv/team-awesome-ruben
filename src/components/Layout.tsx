/* eslint-disable react/jsx-filename-extension */
import React from 'react';

import styles from './Layout.module.css';

interface LayoutProps {
  children: React.ReactNode | React.ReactNode[];
}

const Layout = ({ children }: LayoutProps): React.ReactElement => (
  <main className={styles.layout}>{children}</main>
);

export default Layout;
