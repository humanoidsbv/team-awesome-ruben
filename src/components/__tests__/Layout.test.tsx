import * as React from 'react';
import { shallow } from 'enzyme';

import Layout from '../Layout';

it('matches component output with expected output', () => {
  const layout = shallow(
    <Layout children={<React.Fragment></React.Fragment>} />
  );
  expect(layout).toMatchSnapshot();
});
