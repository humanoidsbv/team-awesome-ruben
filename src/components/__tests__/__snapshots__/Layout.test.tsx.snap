// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches component output with expected output 1`] = `
<main
  className="layout"
/>
`;
